src/main/java/org/quiltmc/loader/impl/transformer/EnvironmentStrippingData.java

```diff
@@ -16,265 +16,142 @@
 
 package org.quiltmc.loader.impl.transformer;
 
+import java.util.ArrayList;
 import java.util.Collection;
-import java.util.HashSet;
 
 import org.objectweb.asm.AnnotationVisitor;
-import org.objectweb.asm.ClassVisitor;
 import org.objectweb.asm.FieldVisitor;
 import org.objectweb.asm.MethodVisitor;
-import org.objectweb.asm.Type;
 import org.objectweb.asm.TypePath;
-import org.objectweb.asm.TypeReference;
-import org.quiltmc.loader.api.minecraft.ClientOnly;
-import org.quiltmc.loader.api.minecraft.DedicatedServerOnly;
 import org.quiltmc.loader.impl.util.QuiltLoaderInternal;
 import org.quiltmc.loader.impl.util.QuiltLoaderInternalType;
 
 import net.fabricmc.api.EnvType;
-import net.fabricmc.api.Environment;
-import net.fabricmc.api.EnvironmentInterface;
-import net.fabricmc.api.EnvironmentInterfaces;
 
-/** Scans a class for Environment and EnvironmentInterface annotations to figure out what needs to be stripped. */
+/** Deprecated. All stuff were moved to {@link StrippingData}. */
+@Deprecated
 @QuiltLoaderInternal(QuiltLoaderInternalType.LEGACY_EXPOSED)
-public class EnvironmentStrippingData extends ClassVisitor {
+public class EnvironmentStrippingData extends StrippingData {
 	// Fabric annotations
-	private static final String ENVIRONMENT_DESCRIPTOR = Type.getDescriptor(Environment.class);
-	private static final String ENVIRONMENT_INTERFACE_DESCRIPTOR = Type.getDescriptor(EnvironmentInterface.class);
-	private static final String ENVIRONMENT_INTERFACES_DESCRIPTOR = Type.getDescriptor(EnvironmentInterfaces.class);
+	private static final String ENVIRONMENT_DESCRIPTOR = StrippingData.ENVIRONMENT_DESCRIPTOR;
+	private static final String ENVIRONMENT_INTERFACE_DESCRIPTOR = StrippingData.ENVIRONMENT_INTERFACE_DESCRIPTOR;
+	private static final String ENVIRONMENT_INTERFACES_DESCRIPTOR = StrippingData.ENVIRONMENT_INTERFACES_DESCRIPTOR;
 
 	// Quilt annotations
-	private static final String CLIENT_ONLY_DESCRIPTOR = Type.getDescriptor(ClientOnly.class);
-	private static final String SERVER_ONLY_DESCRIPTOR = Type.getDescriptor(DedicatedServerOnly.class);
+	private static final String CLIENT_ONLY_DESCRIPTOR = StrippingData.CLIENT_ONLY_DESCRIPTOR;
+	private static final String SERVER_ONLY_DESCRIPTOR = StrippingData.SERVER_ONLY_DESCRIPTOR;
```

Comment:
You don't need to keep any of the *private* fields or classes since external mods can't access them.

Reply:
Just thought that they could be accessed via accessors or somewhat, but now that you're telling me that I don't think mixin can be applied to Quilt Loader classes...